History/HistoryEntry.h

```diff
@@ -61,9 +61,86 @@ struct HistoryInlineRef
     bool        mIsRemote   : 1;
     bool        mIsTag      : 1;
     bool        mIsStash    : 1;
+    bool        mIsDetached : 1;
+
+    bool operator ==(const HistoryInlineRef& other) const
```

Comment:
Why should the compiler fail on that? It will (just like with member-wise copy and member-wise assignment) do a member-wise compare. It will only have problems for more complex operators like less-that, exclusive-or-with-self-assignment etc... :)